dask/dataframe/io/io.py

```diff
@@ -407,6 +407,27 @@ def dataframe_from_ctable(x, slc, columns=None, categories=None, lock=lock):
     return result
 
 
+def _partition_from_array(data, index=None, **kwargs):
+    """Create a Dask partition for either a DataFrame or Series.
+
+    Designed to be used with :func:`dask.blockwise.blockwise`. ``data`` is the array
+    from which the partition will be created. ``index`` can be:
+
+    1. ``None``, in which case each partition has an independent RangeIndex
+    2. a `tuple` with two elements, the start and stop values for a RangeIndex for
+       this partition, which gives a continuously varying RangeIndex over the
+       whole Dask DataFrame
+    3. or an instance of a :class:`dask.dataframe.Index`.
```

Comment:
Correct me if I'm wrong, but I think at this point `index` would be a `pandas.Index`, rather than a `dask.dataframe.Index`